src/vorta/scheduler.py

```diff
@@ -96,21 +114,139 @@ def next_job_for_profile(self, profile_id):
         else:
             return job.next_run_time.strftime('%Y-%m-%d %H:%M')
 
-    def create_backup(self, profile_id):
+    def enqueue_create_backup(self, profile_id, repo_id, vorta_queue):
+        vorta_queue.add_job(CreateJobSched(profile_id=profile_id, site=repo_id))
+
+
+class JobStatus(Enum):
+    # dont add and option to put the item at the end of the queue.
+    OK = 1
+    CANCEL = 2
+
+
+class Job(QObject):
+    """
+    To add a job to the vorta queue, you have to create a class which inherits Job. The inherited class
+    must override run, cancel and get_site_id. Since Job inherits from QObject, you can use pyqt signal.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.__status = JobStatus.OK  # the job can be launched. If False, the job is not run.
+
+    # Keep default
+    def set_status(self, status):
+        self.__status = status
+
+    # Keep default
+    def get_status(self):
+        return self.__status
+
+    # Must return the site id. In borg case, it is the id of the repository.
+    @abstractmethod
+    def get_site_id(self):
```

Comment:
How about calling it repo_id to avoid confusion? Since we don't use the concept of a `site` anywhere else.

Reply:
Yes it's true we can. get_site_id could return other thing that repo id. For example, if in the code I want to run borg version, we can use the manager. Since there is no repo associated with a borg version, get_site_id will just return "borgversion" as repo.

Reply:
Done